pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -59,24 +63,35 @@ public static AggregationFunctionColumnPair getFunctionColumnPair(@Nonnull Aggre
     return new AggregationFunctionColumnPair(functionType, getColumn(aggregationInfo));
   }
 
+  public static boolean isDistinct(AggregationFunctionContext[] functionContexts) {
+    return functionContexts.length == 1 &&
+        functionContexts[0].getAggregationFunction().getType() == AggregationFunctionType.DISTINCT;
+  }
+
   /**
    * Creates an {@link AggregationFunctionContext} from the {@link AggregationInfo}.
    */
   @Nonnull
   public static AggregationFunctionContext getAggregationFunctionContext(@Nonnull AggregationInfo aggregationInfo) {
-    String functionName = aggregationInfo.getAggregationType();
-    AggregationFunction aggregationFunction = AggregationFunctionFactory.getAggregationFunction(functionName);
-    return new AggregationFunctionContext(aggregationFunction, AggregationFunctionUtils.getColumn(aggregationInfo));
+   return getAggregationFunctionContext(aggregationInfo, new BrokerRequest());
```

Comment:
Pass in `null`

Reply:
Done -- although I think making nonnullable would be better. It would then avoid unnecessary conditional check before dereferencing in getAggregationFunction()